packages/ember/tests/routing/router_service_test/refresh_test.js

```diff
@@ -0,0 +1,99 @@
+import { Route } from '@ember/-internals/routing';
+import { RouterTestCase, moduleFor } from 'internal-test-helpers';
+
+moduleFor(
+  'Router Service - refresh',
+  class extends RouterTestCase {
+    async ['@test RouterService#refresh can be used to re-run the model hooks of active routes'](
+      assert
+    ) {
+      let parentCounter = 0;
+      this.add(
+        'route:parent',
+        Route.extend({
+          model() {
+            ++parentCounter;
+          },
+        })
+      );
+
+      let childCounter = 0;
+      this.add(
+        'route:parent.child',
+        Route.extend({
+          model() {
+            ++childCounter;
+          },
+        })
+      );
+
+      let sisterCounter = 0;
+      this.add(
+        'route:parent.sister',
+        Route.extend({
+          model() {
+            ++sisterCounter;
+          },
+        })
+      );
+
+      await this.visit('/');
+      assert.equal(parentCounter, 1);
+      assert.equal(childCounter, 0);
+      assert.equal(sisterCounter, 0);
+
+      await this.routerService.refresh();
+      assert.equal(parentCounter, 2);
+      assert.equal(childCounter, 0);
+      assert.equal(sisterCounter, 0);
+
+      await this.routerService.refresh('application');
+      assert.equal(parentCounter, 3);
+      assert.equal(childCounter, 0);
+      assert.equal(sisterCounter, 0);
+
+      await this.routerService.transitionTo('parent.child');
+      assert.equal(parentCounter, 3);
+      assert.equal(childCounter, 1);
+      assert.equal(sisterCounter, 0);
+
+      await this.routerService.refresh('parent.child');
+      assert.equal(parentCounter, 3);
+      assert.equal(childCounter, 2);
+      assert.equal(sisterCounter, 0);
+
+      await this.routerService.refresh('parent');
+      assert.equal(parentCounter, 4);
+      assert.equal(childCounter, 3);
+      assert.equal(sisterCounter, 0);
+
+      await this.routerService.transitionTo('parent.sister');
+      assert.equal(parentCounter, 4);
+      assert.equal(childCounter, 3);
+      assert.equal(sisterCounter, 1);
+
+      await this.routerService.refresh();
+      assert.equal(parentCounter, 5);
+      assert.equal(childCounter, 3);
+      assert.equal(sisterCounter, 2);
+    }
+
+    async ['@test RouterService#refresh verifies that the provided route exists'](assert) {
+      await this.visit('/');
+
+      assert.throws(() => {
```

Comment:
Is this the best way to test if an assert is thrown?

Reply:
It seems to fail CI, so I'm really starting to think this isn't the way :smile:. Any pointers?

Reply:
I think I found what I was looking for: https://github.com/emberjs/ember.js/blob/22bfcfdac0aeefcf333fb2d6697772934201b43b/packages/internal-test-helpers/lib/ember-dev/assertion.ts#L9